File: hibernate-rediscala/src/test/scala/org/hibernate/cache/rediscala/tests/jpa/JpaAccountRepository.scala
package org.hibernate.cache.rediscala.tests.jpa

import org.springframework.data.jpa.repository.{Query, JpaRepository}
import org.springframework.data.repository.query.Param

/**
 * JpaAccountRepository 
 *
 * @author sunghyouk.bae@gmail.com
 * @since 2014. 2. 26.
 */
trait JpaAccountRepository extends JpaRepository[JpaAccount, java.lang.Long] {

    @Query(value = "select a from JpaAccount a where a.name = :name")
    def findByName(@Param("name") name: String): JpaAccount

}

File: debop4s-data-orm/src/test/java/debop4s/data/orm/jtests/jpa/config/JpaAccountRepository.java
package debop4s.data.orm.jtests.jpa.config;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.transaction.annotation.Transactional;

@Transactional
public interface JpaAccountRepository extends JpaRepository<JpaAccount, Long> {

    @Query(value = "select a from JpaAccount a where a.name = :name")
    JpaAccount findByName(@Param("name") String name);
}
